jetson/percep/artag_detector.cpp

```diff
@@ -127,4 +128,44 @@ cv::aruco::drawDetectedMarkers(rgb, corners, ids);
 
 double TagDetector::getAngle(float xPixel, float wPixel){
     return atan((xPixel - wPixel/2)/(wPixel/2)* tan(fieldofView/2))* 180.0 /PI;
-}
\ No newline at end of file
+}
+
+void TagDetector::updateTag(rover_msgs::Target *arTags, pair<Tag, Tag> &tagPair, Mat &depth_img, Mat &src){
+    struct tagPairs 
+    {
+     vector<int> id;
+     vector<int> locx;
+     vector<int> locy;
+     vector<int> buffer;
```

Comment:
You're mixing 4 spaces for tabs and 2 spaces for tabs (also in a few other locations)